neuromation/api/abc.py

```diff
@@ -99,12 +101,35 @@ class ImageProgressPush:
     dst: RemoteImage
 
 
+@dataclass(frozen=True)
+class ImageProgressSave:
+    job: str
+    dst: RemoteImage
+
+
 @dataclass(frozen=True)
 class ImageProgressStep:
     message: str
     layer_id: str
 
 
+class ImageCommitStatus(str, enum.Enum):
+    STARTED = "CommitStarted"
+    FINISHED = "CommitFinished"
+
+
+@dataclass(frozen=True)
+class ImageCommitDetails:
+    container: str
+    target_image: RemoteImage
+
+
+@dataclass(frozen=True)
+class ImageCommitStep:
```

Comment:
Looks like you have always two `ImageCommitStep` messages: one with `status==CommitStarted`, second with `status=CommitFinished`.

Why not replace it with two distinct message type:
```
@dataclass
class ImageCommitStarted:
    pass  # really no context data? I expect job id at least

@class
class ImageCommitFinished:
    container: str
    target_image: RemoteImage
    # job id again, maybe?
```

Reply:
Honestly, I don't see the difference. Re-worked.